File: Remove Element/remove-element.py
class Solution:
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        i = 0
        n = len(nums)

        while i < n:
            if nums[i] == val:
                nums[i] = nums[n - 1]
                n -= 1
            else:
                i += 1

        return i


solution = Solution()
print(solution.removeElement([3, 2, 2, 3], 3))  # 2
print(solution.removeElement([0, 1, 2, 2, 3, 0, 4, 2], 2))  # 5

File: Remove Element/remove-element.js
/**
 * Remove element of all instances in array
 *
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
const removeElement = (nums, val) => {
  if (!nums.length) {
    return 0;
  }

  let i = 0, n = nums.length;
  while (i < n) {
    if (nums[i] === val) {
      nums[i] = nums[n - 1];
      n--;
    } else {
      i++;
    }
  }

  return i;
};


console.log(removeElement([3, 2, 2, 3], 3)); // 2
console.log(removeElement([0, 1, 2, 2, 3, 0, 4, 2], 2)); // 5